### ISSUE TITLE
access token for every api call

### ISSUE THREAD

**Issue by icegoffy** [AdobeDocs/experience-manager-learn.en#94]
Issue in ./help/forms/aem-forms-with-adobe-campaign/partone.md

so is there any reason why we are getting access token for every api call to campaign?

can't we do refresh token call to adobe io?

**Comment by davidjgonzalez** [comment]
@icegoffy yes - you could leverage Adobe IMS's APIs to refresh the access token. I believe this example excluded that to reduce the code and focus on the interaction between AEM Forms and Campaign. You can add other efficiencies such as caching the token, and re-using the cached token and only refreshing shortly before its expiration (which IIRC, defaults to 24 hours - but you should double-check that and/or that can be overwritten during the exchange w/ IMS)

You can also review some other Adobe IMS code samples for orchestrating this exchange (there's a Java sample too): https://www.adobe.io/authentication/auth-methods.html#!AdobeDocs/adobeio-auth/master/JWT/samples/samples.md
